xarray/core/variable.py

```diff
@@ -1280,6 +1280,30 @@ def stack(self, dimensions=None, **dimensions_kwargs):
             result = result._stack_once(dims, new_dim)
         return result
 
+    def stack_all(self, name='stacked'):
```

Comment:
I'm not sure we need the `xarray.Variable` method.

`Variable` is a lower-level API, mostly for internal use inside xarray to implement functionality for `DataArray` and `Dataset`. If we aren't using a `Variable` method as part of the implementation for Dataset/DataArray, then that is a good clue that it may not be necessary.